plugins/native/module/vtkF3DSPZReader.cxx

```diff
@@ -0,0 +1,288 @@
+  stream.avail_in = compressed.size();
+  if (inflateInit2(&stream, 16 | MAX_WBITS) != Z_OK)
+  {
+    return false;
+  }
+  bool success = false;
+  while (true)
+  {
+    stream.next_out = buffer.data();
+    stream.avail_out = buffer.size();
+    int32_t res = inflate(&stream, Z_NO_FLUSH);
+    if (res != Z_OK && res != Z_STREAM_END)
+    {
+      break;
+    }
+    uncompressed.insert(
+      uncompressed.end(), buffer.data(), buffer.data() + buffer.size() - stream.avail_out);
+    if (res == Z_STREAM_END)
+    {
+      success = true;
+      break;
+    }
+  }
+  inflateEnd(&stream);
+  return success;
+}
+
+//----------------------------------------------------------------------------
+struct Header
+{
+  uint32_t magic;
+  uint32_t version;
+  uint32_t numPoints;
+  uint8_t shDegree;
+  uint8_t fractionalBits;
+  uint8_t flags;
+  uint8_t reserved;
+};
+
+//----------------------------------------------------------------------------
+struct PackedCoordinate
+{
+  uint8_t packed[3]; // 24-bits
+
+  float decode(float scale) const
+  {
+    int32_t fixed32 = packed[0];
+    fixed32 |= packed[1] << 8;
+    fixed32 |= packed[2] << 16;
+    fixed32 |= (fixed32 & 0x800000) ? 0xff000000 : 0; // sign
+    return static_cast<float>(fixed32) * scale;
+  }
+};
+
+//----------------------------------------------------------------------------
+struct ColorChannel
+{
+  uint8_t channel;
+
+  uint8_t decode() const
+  {
+    constexpr float colorScale = 0.15f;
+    constexpr float SH_C0 = 0.28209479177387814;
+
+    float fcolor = ((channel / 255.0f) - 0.5f) / colorScale;
+    fcolor = std::clamp(fcolor * SH_C0 + 0.5, 0.0, 1.0);
+
+    return static_cast<uint8_t>(fcolor * 255.0);
+  }
+};
+
+//----------------------------------------------------------------------------
+struct LogScale
+{
+  uint8_t scale;
+
+  float decode() const
+  {
+    return std::exp((scale / 16.0f - 10.0f));
+  }
+};
+
+//----------------------------------------------------------------------------
+struct PackedRotation
+{
+  uint8_t packed[3]; // 24-bits
+
+  std::array<float, 4> decode() const
+  {
+    float xyz[3] = { static_cast<float>(packed[0]) / 127.5f - 1.f,
+      static_cast<float>(packed[1]) / 127.5f - 1.f, static_cast<float>(packed[2]) / 127.5f - 1.f };
+
+    float w = std::sqrt(std::max(0.0f, 1.0f - vtkMath::Dot(xyz, xyz)));
+
+    return { w, xyz[0], xyz[1], xyz[2] };
+  }
+};
+}
+
+vtkStandardNewMacro(vtkF3DSPZReader);
+
+//----------------------------------------------------------------------------
+int vtkF3DSPZReader::RequestData(
+  vtkInformation*, vtkInformationVector**, vtkInformationVector* outputVector)
+{
+  vtkPolyData* output = vtkPolyData::GetData(outputVector);
+
+  vtkSmartPointer<vtkResourceStream> stream;
+
+#if VTK_VERSION_NUMBER > VTK_VERSION_CHECK(9, 4, 20250501)
+  if (this->Stream)
+  {
+    stream = this->Stream;
+    if (this->Stream->SupportSeek())
+    {
+      this->Stream->Seek(0, vtkResourceStream::SeekDirection::Begin);
```

Comment:
Dont you need to open the file anyway ? superclass only provide the member, but doesnt open  anything.